repositories/convoy_repository.go

```diff
@@ -27,23 +28,89 @@ func getOwnerId(organizationId string, partnerId null.String) string {
 	return fmt.Sprintf("org:%s", organizationId)
 }
 
+// Ensure event type is included in the event data
+func getEventContent(eventContent models.WebhookEventContent) (string, map[string]interface{}) {
+	eventType := string(eventContent.Type)
+	eventData := eventContent.Data
+	eventData["event_type"] = eventType
+	return eventType, eventData
+}
+
 func (repo ConvoyRepository) SendWebhookEvent(ctx context.Context, webhookEvent models.WebhookEvent) error {
 	projectId := repo.convoyClientProvider.GetProjectID()
 	convoyClient, err := repo.convoyClientProvider.GetClient()
 	if err != nil {
 		return err
 	}
 
-	body := convoy.CreateEndpointFanoutEventJSONRequestBody{
-		OwnerId:        utils.Ptr(getOwnerId(webhookEvent.OrganizationId, webhookEvent.PartnerId)),
-		EventType:      utils.Ptr(webhookEvent.EventType.String()),
-		IdempotencyKey: utils.Ptr(webhookEvent.Id),
-		Data:           utils.Ptr(webhookEvent.EventData),
+	ownerId := getOwnerId(webhookEvent.OrganizationId, webhookEvent.PartnerId)
+	eventType, eventData := getEventContent(webhookEvent.EventContent)
+
+	fanoutEvent, err := convoyClient.CreateEndpointFanoutEventWithResponse(ctx, projectId, convoy.ModelsFanoutEvent{
+		OwnerId:        &ownerId,
+		EventType:      &eventType,
+		IdempotencyKey: &webhookEvent.Id,
+		Data:           &eventData,
+	})
+	if err != nil {
+		return errors.Wrap(err, "can't create convoy event: request error")
+	}
+	if fanoutEvent.JSON201 == nil {
```

Comment:
question: how redundant is this with the check of line 55 ? Do we expect a non nil err every time there is no JSON201 ?